sympy/calculus/singularities.py

```diff
@@ -49,22 +50,45 @@ def singularities(expr, sym):
     .. [1] http://en.wikipedia.org/wiki/Mathematical_singularity
 
     """
-    if not expr.is_rational_function(sym):
-        raise NotImplementedError("Algorithms finding singularities for"
-                                  " non rational functions are not yet"
-                                  " implemented")
+    if not expression.is_rational_function(symbol):
+        raise NotImplementedError(
+            "Algorithms finding singularities for non-rational"
+            " functions are not yet implemented."
+        )
     else:
-        return solveset(simplify(1/expr), sym)
+        return solveset(simplify(1 / expression), symbol)
 
 ###########################################################################
 ###################### DIFFERENTIAL CALCULUS METHODS ######################
 ###########################################################################
 
 
-def is_increasing(f, interval=S.Reals, symbol=None):
+def monotonicity_helper(expression, predicate, interval=S.Reals, symbol=None):
+    """
+    Helper function for functions checking function monotonicity.
+
+    It returns whether the interval in which function's derivative
+    satisfies given predicate is superset of given interval.
     """
-    Returns if a function is increasing or not, in the given
-    ``Interval``.
+    expression = sympify(expression)
+
+    if symbol is None:
+        free_variables = expression.free_symbols
+        if len(free_variables) > 1:
+            raise NotImplementedError(
+                'The function has not yet been implemented'
+                ' for all multivariate expressions.'
+            )
+        symbol = free_variables.pop() if len(free_variables) == 1 else Symbol('x')
```

Comment:
Use any variable to differentiate as function is constant anyway - this allows to avoid passing boolean flag whether constant function is (strictly) increasing/decreasing; don't import this variable from `sympy.abc` as this will make doctests fail.